src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISDeployer.cs

```diff
@@ -378,14 +293,96 @@ private void ConfigureAppHostConfig(XElement config, string contentRoot, int por
 
         private void Stop()
         {
-            using (var serverManager = new ServerManager())
+            try
             {
-                var site = serverManager.Sites.SingleOrDefault();
-                site.Stop();
-                var appPool = serverManager.ApplicationPools.SingleOrDefault();
-                appPool.Stop();
-                serverManager.CommitChanges();
+                RetryServerManagerAction(serverManager =>
+                {
+                    var appPool = serverManager.ApplicationPools.SingleOrDefault();
```

Comment:
Shouldn't appPool the order of site and appPool be swapped? Stop the site before stopping the app pool?

Reply:
Good point.